pages/api/collections/[id].js

```diff
@@ -42,15 +42,15 @@ export default async function handler(request, response) {
 
   if (request.method === "DELETE") {
     try {
-      if (session) {
-        await Flashcard.deleteMany({ collectionId: id });
-        await Collection.findByIdAndDelete(id);
-        response.status(200).json({ message: "Collection deleted." });
-        return;
-      } else {
+      if (!session) {
         response.status(401).json({ status: "Not authorized" });
         return;
       }
```

Comment:
this is optional but you could move this check out and in front of the DELETE block. By this you would check the authorization for both the DELETE and the PATCH block and you could save one check.

Reply:
Sounds sensible, I could have thought of that myself. 😉 😅  I've adjusted it here and of course also in the `api/flashcards/[id].js`. Thanks for the tip.